lib/scope.js

```diff
@@ -121,4 +121,28 @@ module.exports = {
 
     return elem;
   },  // Ends getSpecifiedChoice
+
+  detectDownloadComplete: async function detectDownloadComplete(scope, endTime) {
+    /* Resolve if a download flag has changed. Timeout after a bit less than
+    *    the gobal permitted timeout */
+
+    // Stop/resolve if too much time has passed
+    let expirationTime = 1.9 * 60 * 1000;  // m * s * ms
```

Comment:
Not urgent, but I wonder if we should be using a global variable for the permitted timeout, and then this should just be `env_vars.TIMEOUT - 2000` or something.

Reply:
I really like that. I'll make it an issue.